db/migrate/20201116144600_add_validity_to_keypairs.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+class AddValidityToKeypairs < ActiveRecord::Migration[6.0]
+  class Keypair < ActiveRecord::Base; end
+
+  def change # rubocop:disable Metrics/AbcSize, Metrics/MethodLength
+    change_table :keypairs, bulk: true do |t|
+      t.datetime :not_before, precision: 6
+      t.datetime :not_after, precision: 6
+      t.datetime :expires_at, precision: 6
+    end
+
+    reversible do |dir|
+      dir.up do
+        Keypair.find_each do |keypair|
+          keypair.update!(
+            not_before: keypair.created_at,
+            not_after: keypair.created_at + 1.month,
+            expires_at: keypair.created_at + 2.months
+          )
+        end
```

Comment:
Ok, there are probably not many keypairs but this query should be easy to write in SQL

Reply:
Actually, this is not that easy to write in SQL because we need to support both PostgreSQL and SQLite and they don't support the same syntax for adding time to a date. SQLite would be something like `UPDATE keypairs SET not_before = DATE(created_at, '+2629746 seconds')`, while PostgreSQL needs something like `UPDATE keypairs SET not_before = created_at + INTERVAL '2629746 seconds'`.

Since there are very few records, I think doing it like this should be fine.

Reply:
Ahh, that's something I didn't think about 👍